cpp/src/arrow/python/pyarrow_api.h

```diff
@@ -16,16 +16,22 @@
 // under the License.
 
 // DO NOT EDIT THIS FILE. Update from pyarrow/lib_api.h after pyarrow build
+// This is used to be able to call back into Cython code from C++.
 
-/* Generated by Cython 0.29 */
+/* Generated by Cython 0.29.8 */
```

Comment:
why are do we check in this generated code?

Reply:
It has to be installed with the C++ library, right now the C++ build system doesn't know how to invoke Cython. This might be improved later

Reply:
It's a bit more fundamental: these files are generated by Cython and included by our C++ code, but Cython needs the C++ code to be compiled for linking. So there's a circular dependency.